### ISSUE TITLE
Issue with version 10.5.0 and Cypress

### ISSUE THREAD

**Issue by lucho78** [stephanrauh/ngx-extended-pdf-viewer#998]
Hi,
When we run our E2E tests with Cypress the following exception is raised:
`Error: Setting up fake worker failed: "Cannot load script at: http://my-server/assets/pdf.worker.js".`
Do you have an idea how to solve this issue?
Thanks a lot for your help
Regards
Luc

**Comment by aminahasimbegovic** [comment]
Hi! Are there any updates regarding this issue? Any changes with the newest 11.0.0 version?

**Comment by LAlves91** [comment]
Hi @aminahasimbegovic and @lucho78 

Looking through the code, I see this error happens in a `loadScript` function, inside PDF.js...would it be possible for you to change the code a little bit and test it again? The function right now is:

```
function loadScript(src, removeScriptElement = false) {
  return new Promise((resolve, reject) => {
    const script = document.createElement("script");

    if (src.constructor.name === "Function") {
      script.src = src();
    } else {
      script.src = src;
    }

    script.onload = function (evt) {
      if (removeScriptElement) {
        script.remove();
      }

      resolve(evt);
    };

    script.onerror = function () {
      reject(new Error(`Cannot load script at: ${script.src}`));
    };

    (document.head || document.documentElement).appendChild(script);
  });
}
```
If you change the onerror function, I think we'll have a better understanding of what's happening:
```
    script.onerror = function (error) {
      console.log(error);
      reject(new Error(`Cannot load script at: ${script.src}`));
    };
```
Thank you!

**Comment by stephanrauh** [comment]
@LAlves91 Yeah, sure. If it help you to help me, that's a win-win situation! :)

Your additional log statement has landed with version 11.0.0-alpha.2. Note that I've only added it to the "bleeding edge" branch.

BTW, I didn't expect the bug to lurk inside pdf.js. But it makes sense. pdf.js is evolving to quickly I don't manage to keep track of all the changes. I just wonder if it's possible to convince Cypress to use the methods `loadViewer()` and `loadPdfJs()` of the `NgxExtendedPdfViewerComponent`?

**Comment by LAlves91** [comment]
Hi @stephanrauh 

Sorry if I didn't express it correctly (I write weirdly sometimes in english), but I didn't mean to ask you to change the lib's code! I was asking the creator of this issue to just add that statement in his project and run his tests, to see what was actually happening!

I don't have experience with cypress, but my first thought was that his test could be setting the worker's path in a way it couldn't be reached in the testing environment. So logging the error could give some clarity on what made the loading fail.

**Comment by stephanrauh** [comment]
@LAlves91 OK :) However, I think it makes sense not to swallow the error message, so I think my change is useful.

**Comment by LAlves91** [comment]
Absolutelly! It'll improve bug diagnostics a lot!

**Comment by stephanrauh** [comment]
@LAlves91 @lucho78 @aminahasimbegovic Can anybody help me to find out what's going wrong? When I start Cypress, my laptop is eating battery like nothing good, and it stops being responsive. That makes it hard to find out what's going wrong. 

It seems that the global option `workerSrc` isn't set, so the application falls back to a default behavior that's broken since ages. I could repair the default behavior, but I'd prefer to fix the root problem.

**Comment by steve-accelex** [comment]
viewer-2.13.265.js  specifically sets the workerSrc on line 232

  workerSrc: {
    value: "./assets/pdf.worker.js",
    kind: OptionKind.WORKER
  }

This is then called in Load Fake worker on line 2235

async function loadFakeWorker() {
  if (!_pdfjsLib.GlobalWorkerOptions.workerSrc) {
    _pdfjsLib.GlobalWorkerOptions.workerSrc = _app_options.AppOptions.get("workerSrc");

    if (_pdfjsLib.GlobalWorkerOptions.workerSrc.constructor.name === "Function") {
      _pdfjsLib.GlobalWorkerOptions.workerSrc = _pdfjsLib.GlobalWorkerOptions.workerSrc();
    }
  }

  await (0, _pdfjsLib.loadScript)(_pdfjsLib.PDFWorker.workerSrc);
}

So there is a reference here to a file that no longer exists in the source

**Comment by stephanrauh** [comment]
As far as I remember, the problem is really related with the `workerSrc` attribute. It is used, but it should have a completely different value. It seem Cypress initializes the Application differently. The line that sets `workerSrc` correctly is never executed. Probably the constructor or the method `ngOnInit()` of the `NgxExtendedPdfViewerComponent` is called to0 late.

**Comment by stephanrauh** [comment]
The other question is: why is `loadFakeWorker()` called at all? The real browser never calls this method (unless you add the worker to the scripts section of the `angular.json`, but that turned out to be a bad solution).

**Comment by stephanrauh** [comment]
Found it! Cypress embeds the PDF viewer in an iFrame. Unfortunately, this means that the `webviewerloaded` event is not sent to ngx-extended-pdf-viewer, but to Cypress:

```javascript
      // Give custom implementations of the default viewer a simpler way to
      // set various `AppOptions`, by dispatching an event once all viewer
      // files are loaded but *before* the viewer initialization has run.
      const event = document.createEvent("CustomEvent");
      event.initCustomEvent("webviewerloaded", true, true, {
        source: window,
      });
      try {
        // Attempt to dispatch the event at the embedding `document`,
        // in order to support cases where the viewer is embedded in
        // a *dynamically* created <iframe> element.
        parent.document.dispatchEvent(event);
      } catch (ex) {
        // The viewer could be in e.g. a cross-origin <iframe> element,
        // fallback to dispatching the event at the current `document`.
        Window['ngxConsole'].error(`webviewerloaded: ${ex}`);
        document.dispatchEvent(event);
      }
```

**Comment by stephanrauh** [comment]
The bugfix has landed with version 11.1.2.
